WireUI/Sources/WireAccountImage/UseCases/GetTeamAccountImageUseCaseProtocol.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+/// If the team has an image set, this use cases retrieves it.
+/// If no team image data is available, an image will be generated using the team name's initials.
+public protocol GetTeamAccountImageUseCaseProtocol {
+
+    func invoke<User, Account>(user: User, account: Account) async throws -> UIImage
```

Comment:
How come you use generics here instead of 

```swift
 func invoke(
    user: GetAccountImageUseCaseUserProtocol, 
    account: GetAccountImageUseCaseAccountProtocol
) async throws -> UIImage
```

Reply:
Because I prefer static dispatching.
Actually I forgot to change it to the compact syntax using `some` in the two protocols.
Thanks for pointing that out.
Fixed: 3ef1d0cfc236667efc07a833cf416ccda1a71f81

Reply:
OK I see. Yeah I think this is worthwhile and the direction Swift is pushing us. Perhaps this is worth some discussion (maybe it has been discussed already) so have this as a principle in our project